resolvers/__tests__/context.spec.js

```diff
@@ -0,0 +1,83 @@
+const { setContext } = require('../context')
+const { fakeCookie } = require('../stubs')
+const { User } = require('../../models/User')
+
+afterEach(() => {
+  jest.clearAllMocks()
+})
+
+beforeEach(async () => {
+  await User.query().delete()
+})
+
+const getAllUsersCount = async () => {
+  const usersCount = await User.query().count('id')
+
+  return Number(usersCount[0].count)
+}
+
+test('returns an object with empty user when no auth cookie found', async () => {
+  const consoleError = jest.fn()
+  jest.spyOn(console, 'error').mockImplementation(consoleError)
+  const req = {
+    headers: {
+      cookie: ''
+    }
+  }
+
+  const context = await setContext({ req })
+
+  expect(context).toEqual({ User: {} })
+  expect(consoleError).toHaveBeenCalledTimes(1)
+})
+
+test('returns an object with user when auth cookie is found', async () => {
+  const consoleError = jest.fn()
+  jest.spyOn(console, 'error').mockImplementation(consoleError)
+  const req = {
+    headers: {
+      cookie: fakeCookie
+    }
+  }
+
+  const context = await setContext({ req })
+
+  expect(context.User).toEqual(
+    expect.objectContaining({
+      external_id: 3,
+      username: 'alexis1',
+      email: 'alexis@mail.com'
+    })
+  )
+  expect(consoleError).not.toHaveBeenCalledTimes(1)
+})
+
+test('creates a new db record with user data if user is not present', async () => {
+  const req = {
+    headers: {
+      cookie: fakeCookie
+    }
+  }
+  const prevAmountUsers = await getAllUsersCount()
```

Comment:
I think we can leave the query here instead of moving it to a method. It will make the test easier to read

Reply:
I will argue that I have it four times and it is kinda pain to repeat all. Why do you want to see that method? It doesn't have anything to do with the test goal itself for reader is just _"I know we need to count the users"_

Reply:
That method is expected for you as reader to be as a blackbox that give you the count of the users.

Reply:
The tests shouldn't be treated as black boxes. We are obscuring how we are getting this count unnecessarily.

Also about repetition, you are just moving 1 line of code to a method, I don't see how that would be painful.